src/arch/arm/subarch/armv7a/cpu_reset.S

```diff
@@ -13,6 +13,12 @@
 
 .global hardware_init_hook
 hardware_init_hook:
+	/* check primary core */
+	mrc     p15, 0, r0, c0, c0, 5
+	ubfx    r0, r0, #0, #2
+	cmp     r0, #0
+	bne		halt
```

Comment:
It's a common code for armv7a not only for RasPi. Could you modify it in a more common way? Maybe use option and `#ifdef` or something else.